Improve inbox activity type handling and tests

pfefferle opened the pull request:
Replaces strtolower with camel_to_snake_case for activity and object type normalization in the inbox handler. Updates tests to cover additional cases, including QuoteRequest activities and handling of non-array objects, ensuring more robust and flexible activity type processing.

## Proposed changes:
- Replaces `strtolower` with `camel_to_snake_case` for consistent type normalization
- Adds handling for non-array objects in inbox processing
- Expands test coverage with QuoteRequest activities and additional edge cases

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* See unittests

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [X] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [X] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Fixed activity processing to handle QuoteRequest and other edge cases more reliably.

</details>

pfefferle committed:
Improve inbox activity type handling and tests

Replaces strtolower with camel_to_snake_case for activity and object type normalization in the inbox handler. Updates tests to cover additional cases, including QuoteRequest activities and handling of non-array objects, ensuring more robust and flexible activity type processing.

pfefferle committed:
Fix handling of empty object type in inbox handler

Updated the inbox handler to return failure when the object type is empty or not in the allowed list. Adjusted the related test to expect failure for Create activity with an empty object type.

pfefferle committed:
Remove all filters for inbox activity types in test

Added a call to remove_all_filters for 'activitypub_persist_inbox_activity_types' in the Test_Inbox test case to ensure a clean test environment and prevent side effects from lingering filters.

matticbot committed:
Add changelog

pfefferle committed:
Merge branch 'trunk' into fix/inbox-handler

obenland commented on includes/handler/class-inbox.php:
```diff
@@ -56,9 +56,11 @@ public static function handle_inbox_requests( $data, $user_id, $type, $activity
 
-			if ( isset( $data['object']['type'] ) && ! \in_array( \strtolower( $data['object']['type'] ), $object_types, true ) ) {
+			if ( ! is_array( $data['object'] ) ) {
+				$success = true;
```
At this point `$success` is already true and `$object_types` are unused. Maybe line 52 should be 
```php
// A comment that explains what's being tested for.
if ( $success && is_array( $data['object'] ) ) {
```

pfefferle committed:
Merge branch 'trunk' into fix/inbox-handler

pfefferle committed:
Fix object type validation logic in inbox handler

Corrects the conditional to only set $success to false when the object is an array and its type is missing or not allowed. Previously, non-array objects were incorrectly marked as successful.

pfefferle committed:
Merge branch 'trunk' into fix/inbox-handler

pfefferle committed:
Merge branch 'trunk' into fix/inbox-handler
